added arangodb backend

dilipvamsi opened the pull request:
## ArangoDB Backend.
### Backend url
_**arangodb://username:password@host:port/database/collection**_
### Custom Backend settings 
**arangodb_backend_settings** is where the settings are present in the app.conf.
Settings should contain the **host, port, username, password, database name, collection name** else the default will be chosen. 
## Task Document in DB
Task is written into the collection with data in task attribute and task_id as _key (document key)
### Sample Document
```json
  {
    "_key": "celery-task-meta-820dd23a-d16e-4527-b15b-aaeb145857bf",
    "_id": "celery/celery-task-meta-820dd23a-d16e-4527-b15b-aaeb145857bf",
    "_rev": "_X1isPje--_",
    "task": {
      "status": "SUCCESS",
      "result": 7,
      "traceback": null,
      "children": [],
      "task_id": "820dd23a-d16e-4527-b15b-aaeb145857bf",
      "date_done": "2018-12-04T12:25:34.709131"
    }
  }
```

Regards,
Dilip Vamsi.

clokep commented on celery/backends/arangodb.py:
```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+"""Couchbase result store backend."""
```
Is this a copy and paste error? (Couchbase vs. ArangoDB?)

dilipvamsi replied:
Yes.

dilipvamsi commented:
Hi @clokep ,
Can you help me for the required
-  Test Cases
-  Proper Documentation for integration with existing docs.

Thanks,
Dilip Vamsi.

clokep commented on celery/app/backends.py:
```diff
@@ -37,6 +37,7 @@
     'consul': 'celery.backends.consul:ConsulBackend',
     'dynamodb': 'celery.backends.dynamodb:DynamoDBBackend',
     'azureblockblob': 'celery.backends.azureblockblob:AzureBlockBlobBackend',
+    'arangodb': 'celery.backends.arangodb.ArangoDbBackend'
```
I think you need a `:` before `ArangoDbBackend` instead of a `.`.

Also, please add a trailing comma here.

dilipvamsi replied:
I didn't get the first comment

dilipvamsi replied:
I have added the trailing comma

clokep replied:
@dilipvamsi You have `celery.backends.arangodb.ArangoDbBackend` (everything separated by dots), but if you look at the other backends, the class-name (e.g. `ArangoDbBackend`) is separated by the module via a colon, so it should be `celery.backends.arangodb:ArangoDbBackend`. Subtle, but I believe it is what lets the dynamic importing work properly.

clokep commented:
@dilipvamsi I've not added a backend, so not 100% sure what test-cases are necessary! Sorry. I know there was a couple of other PRs out recently about this though. You might derive inspiration from them! For docs, I'd look at the same thing (or try to make similar documentation as what exists for other backends).

thedrow commented on celery/backends/arangodb.py:
```diff
@@ -0,0 +1,196 @@
+                database_collection, _query
+            ) = _parse_url(url)
+            logging.debug(
+                "_shema: %s, host: %s, port: %s, username: %s, password: %s, "
```
Never log passwords.
I'd rather remove that line entirely anyway.

thedrow commented on celery/backends/arangodb.py:
```diff
@@ -0,0 +1,196 @@
+from .base import KeyValueStoreBackend
+
+try:
+    from pyArango import connection as pyArangoConnection
```
Module name should be name_with_underscores.

dilipvamsi commented:
Hi @thedrow ,
Can you explain about the `bandit.json` file and whats needed to be done in that file.
Thanks,
Dilip Vamsi

dilipvamsi commented:
Hi, @thedrow ,
I have taken example of couchbase and written the documentation. But it seems that I'm missing something as sphinx build is failing. Can you look into it and help me for the required.

Thanks,
Dilip Vamsi.

thedrow commented on celery/backends/arangodb.py:
```diff
@@ -60,7 +60,7 @@ def __init__(self, url=None, *args, **kwargs):
                 database_collection, _query
             ) = _parse_url(url)
             logging.debug(
```
Why do we need this log line?

dilipvamsi replied:
Its just the configuration of the backend.
Are there any strict guidelines regarding the logs as I'm even logging on a get, set, mget and delete

dilipvamsi commented:
Hi @thedrow @clokep @georgepsarakis @auvipy,
I have fixed the sphinx documentation and added the arangodb test cases (all passing when I locally ran them).
But Travis CI seems to fail because of the setup error as shown below.
Please update and let me know.
![setup error](https://user-images.githubusercontent.com/16288600/52851983-6e9c1680-313d-11e9-8ee4-9238d31b40a7.png)

Thanks,
Dilip Vamsi.

thedrow commented:
If you could, please add integration tests.

dilipvamsi commented:
Hi @thedrow @auvipy ,
Can you suggest me what needs to be implemented in integration testing.
Thanks,
Dilip Vamsi.

thedrow commented:
We just need to run ArangoDB in CI.
So include it in tox and in the build matrix.
